### PR TITLE
add inf cv notebooks

### PR DESCRIPTION
**Pull request by bogdan-galileo**
- Add inference cells on CV notebooks, EXCEPT the one used for the example runs (HF)
- Rename CV notebooks to 
   - HF_data for HuggingFace
   - DF_data for DataFrame
   - IF_data for ImageFolder
- Add all CV notebooks to CI + add minimize_for_CI

### CODE REVIEW

File: .github/workflows/examples.yaml [hunk 1]
```diff
@@ -152,7 +152,7 @@ jobs:
           GALILEO_MODAL_TOKEN_SECRET: ${{ secrets.GALILEO_MODAL_TOKEN_SECRET }}
         run: |
           modal token set --token-id ${{ env.GALILEO_MODAL_TOKEN_ID }} --token-secret ${{ env.GALILEO_MODAL_TOKEN_SECRET }}
-          modal run ./scripts/populator_cuda.py::run --notebook-path examples/image_classification/Image_Classification_using_PyTorch_and_🔭_Galileo.ipynb
+          modal run ./scripts/populator_cuda.py::run --notebook-path examples/image_classification/Image_Classification_using_PyTorch_and_🔭_Galileo_IF_data.ipynb
```
Comment (anthonycorletti) [inline .github/workflows/examples.yaml]: this should not be removed because its for example runs.

File: .github/workflows/examples.yaml [hunk 2]
```diff
@@ -258,7 +258,7 @@ jobs:
           SLACK_WEBHOOK: ${{ secrets.ML_TEAM_EXAMPLES_WEBHOOK }}
           SLACK_USERNAME: "Example Runs Root Content Populator"
           SLACK_ICON_EMOJI: ":telescope:"
-          SLACK_COLOR: ${{ ((needs.mltc-example.result == 'success') && (needs.ner-example.result == 'success') && (needs.tc-example.result == 'success') && (needs.ic-example.result == 'success') && (needs.od-example.result == 'success') && (needs.semseg-example.result == 'success') && 'success') || 'failure' }}
+          SLACK_COLOR: ${{ ((needs.mltc-example.result == 'success') && (needs.ner-example.result == 'success') && (needs.tc-example.result == 'success') && (needs.ic-example-HF.result == 'success') && (needs.od-example.result == 'success') && (needs.semseg-example.result == 'success') && 'success') || 'failure' }}
```
Comment (anthonycorletti) [inline .github/workflows/examples.yaml]: same here, should be changed back. if there's an inference example run we can add a separate job in the dag.

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -44,7 +44,9 @@ jobs:
       tc-keras: ${{ steps.filter.outputs.tc-keras }}
       tc-pytorch: ${{ steps.filter.outputs.tc-pytorch }}
       tc-tensorflow: ${{ steps.filter.outputs.tc-tensorflow }}
-      ic-pytorch: ${{ steps.filter.outputs.ic-pytorch }}
```
Comment (anthonycorletti) [inline .github/workflows/test.yaml:47]: why is this deleted?

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -646,10 +652,10 @@ jobs:
       - name: run notebook
         run: jupyter nbconvert --to notebook --execute examples/text_classification/Text_Classification_using_Tensorflow_and_🔭_Galileo.ipynb
 
-  ic-pytorch:
+  ic-pytorch-HF:
     runs-on: ubuntu-latest
     needs: paths-filter
-    if: ${{ github.event_name != 'pull_request' || (github.event_name == 'pull_request' && needs.paths-filter.outputs.ic-pytorch == 'true' ) }}
```
Comment (anthonycorletti) [inline .github/workflows/test.yaml:652]: should not be deleted. copy paste errors?

**Review by anthonycorletti — CHANGES_REQUESTED** [review]
@bogdan-galileo instead of overwriting the image classification examples, could we have a separate dir for inference? i.e. `examples/image_classification_inference` or `examples/image_classification/inference`?

File: .github/workflows/test.yaml (hunk 3 above)
Comment (bogdan-galileo) [reply .github/workflows/test.yaml:47]: @anthonycorletti we only had 1 notebook being tested ic-pytorch, but we have 3 IC notebooks, all pytorch. So I just added the missing 2 and renamed appropriately

File: .github/workflows/examples.yaml (hunk 1 above)
Comment (bogdan-galileo) [reply .github/workflows/examples.yaml]: @anthonycorletti I didn't add an inference notebook by itself, I just added inference cells at the end of the normal notebooks. Also, renamed our notebooks since the names were a bit weird, it's now
- Galileo_HF_data for HuggingFace
- Galileo_IF_data for ImageFolder
- Galileo_DF_data for DataFrame

Comment (bogdan-galileo) [reply .github/workflows/examples.yaml]: I see, these will now fail in cloud since they have inference. Okay will fix that

**Review by anthonycorletti — APPROVED** [review]
looks good! ready to go whenever the tests are.
